kyo-core/shared/src/main/scala/kyo/hubs.scala

```diff
@@ -74,17 +77,16 @@ class Hub[T: Flat] private[kyo] (
     def close: Option[Seq[T]] < IOs =
         fiber.interrupt.map { _ =>
             ch.close.map { r =>
-                val it = listeners.iterator()
-                def loop(): Unit < IOs =
-                    IOs {
-                        if it.hasNext() then
-                            IOs.attempt(it.next().close)
-                                .map(_ => loop())
+                IOs {
```

Comment:
This `IOs` suspension was missing here since it deals with a mutable collection. I've not used `iterator` to avoid mutable state between the transformations.